src/logics/Logic.cc

```diff
@@ -944,6 +944,27 @@ PTRef Logic::mkEq(vec<PTRef>& args) {
     return mkFun(eq_sym, args);
 }
 
+PTRef Logic::mkDistinct(vec<PTRef>& args) {
+    if (args.size() == 0) return getTerm_false(); // Is this correct?
```

Comment:
If I read SMT-LIB standard correctly, distinct is defined only for 2 or more arguments, so distinct term with 0 or 1 argument is probably an error.
Nevertheless, I would argue that both distinct over 1 and 0 arguments should be TRUE, since in both cases, all arguments are distinct (the case of 0 arguments is a claim with universal quantification over empty set, so should be automatically true).

Reply:
> I would argue that both distinct over 1 and 0 arguments should be TRUE, since in both cases, all arguments are distinct

Yes.  I'm adding a definition to the comments:

  Given args = {a_1, ..., a_n}, distinct(args) holds iff for all a_i, a_j \in args s.t. i != j: a_i != a_j

This hopefully is consistent what you wrote there.  It sounds reasonable to me.

>  distinct term with 0 or 1 argument is probably an error.

The parser already seems to throw an error.  But maybe the API can allow this, so I agree that it should be fixed.